### ISSUE TITLE
False positive findings for rule [SAY_NO_TO_VAR]

### ISSUE THREAD

**Issue by WebTiger89** [saveourtool/diktat#1327]
> [SAY_NO_TO_VAR] Usage of a mutable variables with [var] modifier - is a bad style, use [val] instead: var result = name.hashCode()

This code snippet is generated by Android:
```
    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + address.hashCode()
        result = 31 * result + bondState
        result = 31 * result + rssi
        return result
    }
```

**Comment by akuleshov7** [comment]
Unfortunately we won't be able to determine all cases of such code :( Generally, the usage of `var` is a bad practice, except some corner cases where performance is needed. For these cases we have `Suppress` mechanism and possibility to disable inspection in the config.

**Comment by WebTiger89** [comment]
This case would benefit from an excludes property which I suggested here: https://github.com/saveourtool/diktat/issues/1325

**Comment by WebTiger89** [comment]
Just my opinion but it seems you compensate missing core features with @Suppress. Imagine you have @Suppress annotations in your whole code base with thousands of files and suddenly decide to not use Diktat anymore or Diktat becomes abandoned. You would have to touch every file to remove these useless annotations. That is not the right way, it's bad practice and a code smell.

**Comment by WebTiger89** [comment]
@akuleshov7 Hmm the I have to ask what you mean with

> For these cases we have Suppress mechanism

On the main page I can only find @Supress annotations. Isn't that what you meant?
**Sorry for answering to a closed issue. If this is a problem, just let me know and I will avoid it.**
